src/server/transaction.cc

```diff
@@ -618,7 +623,7 @@ bool Transaction::RunInShard(EngineShard* shard, bool txq_ooo) {
   // For multi we unlock transaction (i.e. its keys) in UnlockMulti() call.
   // If it's a final hop we should release the locks.
   if (is_concluding) {
-    bool became_suspended = sd.local_mask & SUSPENDED_Q;
+    bool became_suspended = sd.local_mask & WAS_SUSPENDED;
```

Comment:
I think that it will be a little bit more clear to write
became_suspended = !was_suspended && sd.local_mask & WAS_SUSPENDED
and the if statement in line 640 will be if (!became_suspended)